opm/parser/eclipse/IntegrationTests/TransMultIntegrationTests.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+  Copyright 2013 Statoil ASA.
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#define BOOST_TEST_MODULE TransMultTests
+#include <math.h>
+
+#include <boost/test/unit_test.hpp>
+#include <boost/test/test_tools.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/Parser/Parser.hpp>
+#include <opm/parser/eclipse/Parser/ParseMode.hpp>
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/ScheduleEnums.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/CompletionSet.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Events.hpp>
+#include <opm/parser/eclipse/Units/ConversionFactors.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/GroupTreeNode.hpp>
+
+using namespace Opm;
+
+
+BOOST_AUTO_TEST_CASE(MULTFLT_IN_SCHEDULE) {
+    ParserPtr parser(new Parser());
+    boost::filesystem::path scheduleFile("testdata/integration_tests/TRANS/Deck1");
+    DeckPtr deck =  parser->parseFile(scheduleFile.string(), parseMode);
+    std::shared_ptr<EclipseState> eclState = std::make_shared<EclipseState>( deck , parseMode );
+    std::shared_ptr<const TransMult> trans = eclState->getTransMult();
+    std::shared_ptr<const Schedule> schedule = eclState->getSchedule();
+    const Events& events = schedule->getEvents();
+
+    BOOST_CHECK_EQUAL( 0.10 , trans->getMultiplier( 3,2,0,FaceDir::XPlus ));
+    BOOST_CHECK_EQUAL( 0.10 , trans->getMultiplier( 2,2,0,FaceDir::XPlus ));
```

Comment:
On Tue, Nov 03, 2015 at 01;2802;0c3:19:25AM -0800, Joakim Hove wrote:
> 
> > would it not be better if the application of modifiers happened inside the EclipseState?
> 
> Yes - that is certainly possible. The main point as I see it is that calling scope, e.g. `flow` must be responsible for checking for events - and then ask the EclipseState to recalculate. Either as here by calling an `apply()`method or asking for a new version of the `TransMult` field - I guess the latter is cleaner.
> 

Currently the eclipse parser works on the whole grid while flow might
only use a part of it (e.g. in a parallel run). If flow calls to the
eclipse parser to recompute something, then this should not need to
happen on the whole grid. Otherwise it does not scale.